engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import com.google.common.collect.Maps;
+import org.terasology.assets.ResourceUrn;
+
+import java.util.Map;
+
+public class FlexibleConfig {
+    private Map<Key<?>, Setting> settingMap;
+
+    public FlexibleConfig() {
+        this.settingMap = Maps.newHashMap();
+    }
+
+    public <V> Key<V> add(Setting<V> setting) {
+        Key<V> key = new Key<>(setting.getId(), setting.getValue());
+
+        // Maybe throw an exception?
+        if (has(key))
+            return null;
+
+        settingMap.put(key, setting);
+        return key;
+    }
+
+    public boolean remove(Key<?> id) {
+        Setting setting = get(id);
+
+        if (setting == null || setting.hasSubscribers())
+            return false;
+
+        settingMap.remove(id);
+        return true;
+    }
+
+    @SuppressWarnings("unchecked")
+    public <V> Setting<V> get(Key<V> key) {
+        if (key == null || key.vClass == null) return null;
+
+        Setting setting = settingMap.get(key);
+
+        if (setting == null) return null;
+
+        // Possibly raise an exception?
```

Comment:
It would be defined as 
```java
<V> Setting<V> get(SimpleUri uri, Class<V> valueClass) {
    return (Setting<V>) settingMap.get(uri);
}
```
Since the `Class<V>` will not be used (unless you want to return `null` instead of causing a class cast exception when the types do not match), the parameter can be removed. That's why I prefer `config.get<Integer>(uri)` instead of this signature. Plus, `get<Integer>(uri)` makes it more clear than `get(uri, Integer.class)` that the fetched type will be a `Setting<Integer>`, since the class parameter could be used for absolutely anything.